[CL-3581] Implement "/api/v2/<collection>/deleted" endpoints (public API)

adessy opened the pull request:
# Changelog
<!-- Replace this comment by a bullet list. More info: https://www.notion.so/citizenlab/Changelog-How-it-works-f418426c75994454a332bf067634f3f1 -->

adessy committed:
[CL-3581] Implement basic version of the `deleted` endpoint for ideas (public API)

adessy committed:
[CL-3581] Rewrite the `deleted` route using a concern

adessy committed:
[CL-3581] Refactor "deleted" action as a reusable concern

This commit implements the "deleted" action as a concern, allowing for its reuse in other controllers.

adessy committed:
[CL-3581] Implement query parameter for filtering deleted items based on deletion date

adessy committed:
[CL-3581] Refactor 'deleted' endpoint tests using shared examples

adessy committed:
[CL-3581] Split shared.rb into multiple files

adessy committed:
[CL-3581] Add a 'deleted' endpoint for all resources

adessy committed:
[CL-3581] Fix rubocop offenses

jamesspeake reviewed (approved):
Looks good to me. Good reorganisation of the shared examples too. One thing that might be worth checking if you haven't already is whether all these models do actually create a 'deleted' activity.

adessy committed:
[CL-3581] Remove the deleted endpoint for reactions

This endpoint is useless for reactions because we don't currently log "deleted" activities for reactions.
